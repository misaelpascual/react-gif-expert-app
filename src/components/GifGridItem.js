import React from 'react'
import PropTypes from 'prop-types'

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__zoomIn animate__delay-1s">
      <img src={ url } alt={ title } />
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string,
  url: PropTypes.string.isRequired
}