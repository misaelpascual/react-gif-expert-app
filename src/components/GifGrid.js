import React from 'react'
import PropTypes from 'prop-types'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
import { LoadingBase } from './UI/LoadingBase'

export const GifGrid = ({ category }) => {

  // const [images, setImages] = useState([]);
  const { data:images, loading } = useFetchGifs(category)


  return (
    <>
      <h3 className='animate__animated animate__fadeInDown animate__delay-1s'> { category } </h3>
      { loading && <LoadingBase /> }
      <div className="card_grid">
        {
          images?.map((img) => (
            <GifGridItem
              key={ img.id }
              { ...img }
            />
          ))
        }
      </div>
    </>
  )
}

GifGrid.protoTypes = {
  category: PropTypes.string.isRequired
}
