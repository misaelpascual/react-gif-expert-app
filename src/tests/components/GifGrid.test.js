import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')

describe('Pruebas en el componente <GifGrid />', () => {

  const category = 'Cats'

  test('Debe mostrar <GifGrid /> correctamente', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })

    const wrapper = shallow(<GifGrid category={category} />)

    expect(wrapper).toMatchSnapshot()
  })


  test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {
    const gifs = [{
        id: 1,
        url: 'https://example.com/test/img.gif',
        title: 'Titulo'
    }]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    })

    const wrapper = shallow(<GifGrid category={category} />)

    expect(wrapper).toMatchSnapshot()

    // Validamos que no exista la palabra loading en un parrafo
    // expect(wrapper.find('p').exists()).toBe(false)

    // Validamos que no exista el componente LoadingBase
    expect(wrapper.find('LoadingBase').exists()).toBe(false)
    // Validamos que exista el GifGridItem
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
  })

})