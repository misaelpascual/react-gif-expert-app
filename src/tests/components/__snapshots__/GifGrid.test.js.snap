// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <GifGrid /> Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown animate__delay-1s"
  >
     
    Cats
     
  </h3>
  <div
    className="card_grid"
  >
    <GifGridItem
      id={1}
      key="1"
      title="Titulo"
      url="https://example.com/test/img.gif"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el componente <GifGrid /> Debe mostrar <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown animate__delay-1s"
  >
     
    Cats
     
  </h3>
  <LoadingBase />
  <div
    className="card_grid"
  />
</Fragment>
`;
